meinberlin/react/account/FeedList.jsx

```diff
@@ -0,0 +1,182 @@
+    (page - 1) * PAGE_SIZE,
+    page * PAGE_SIZE
+  )
+  const unreadCount = notifications.filter(
+    (notification) => !notification.read
+  ).length
+
+  useEffect(() => {
+    const fetchAllNotifications = async () => {
+      try {
+        setLoading(true)
+        setError(null)
+
+        let allNotifications = []
+        let currentPage = 1
+        let totalCount = 0
+
+        do {
+          const response = await fetch(
+            apiUrl + '?page=' + currentPage + '&page_size=50'
+          )
+
+          if (!response.ok) {
+            throw new Error(translations.errorNotificationsText)
+          }
+
+          const data = await response.json()
+
+          if (currentPage === 1) {
+            totalCount = data.count
+          }
+
+          allNotifications = [...allNotifications, ...data.results]
+          currentPage++
+        } while (allNotifications.length < totalCount)
+
+        setNotifications(allNotifications)
+      } catch (err) {
+        setError(err.message)
+      } finally {
+        setLoading(false)
+      }
+    }
+
+    fetchAllNotifications()
+  }, [])
+
+  const handleMarkAllAsRead = async () => {
+    try {
+      setError(null)
+
+      const response = await updateItem({ read: true }, apiUrl + '?page_size=' + notifications.length, 'POST')
+
+      if (!response.ok) {
+        throw new Error(translations.errorNotificationsText)
+      }
+
+      const data = await response.json()
+      setNotifications(data.results)
+    } catch (err) {
+      setError(err.message)
+    }
+  }
+
+  return (
+    <div className="feed-list" aria-live="polite">
+      {loading
+        ? <Spinner />
+        : error
+          ? <Alert type="danger" message={translations.errorText + ': ' + error} />
+          : (
+            <>
+              <div className="feed-list__text">
+                <div>
+                  <h2 className="feed-list__title">
+                    {title}{' '}
+                    {unreadCount > 0 && (
+                      <span className="feed-list__count">{unreadCount}</span>
+                    )}
+                  </h2>
+                  <p className="feed-list__description">
+                    {notifications.length > 0 ? description : descriptionNoItems}
+                  </p>
+                  {notifications.length === 0 && <a className="button" href={link}>{buttonText}</a>}
+                </div>
+                {unreadCount > 0 && (
+                  <div className="feed-list__mark-as-read">
+                    <button className="feed-list__mark-as-read-button" onClick={handleMarkAllAsRead}>
+                      <i className="fa-solid fa-list-check mr-1" />
+                      {translations.markAllAsRead}
+                    </button>
+                  </div>
+                )}
+              </div>
+              {paginatedNotifications.map((notification, index) =>
+                renderFeedItem(notification, index)
+              )}
```

Comment:
Can you put this into a `ul > li` construct? Maybe have the `li` inside the list component and not the `FeedItem` so that it's more modular.